src/atc/orchestrators/ehjson2delta/EhJsonToDeltaExtractor.py

```diff
@@ -0,0 +1,118 @@
+from datetime import datetime as dt
+from datetime import timezone
+from typing import Optional
+
+from pyspark.sql import DataFrame
+from pyspark.sql import functions as f
+
+from atc.delta import DeltaHandle
+from atc.eh.EventHubCaptureExtractor import EventHubCaptureExtractor
+from atc.etl import Extractor
+from atc.orchestrators.ehjson2delta import EhJsonToDeltaException
+from atc.spark import Spark
+
+
+class EhJsonToDeltaExtractor(Extractor):
+    """
+    This extractor has a side effect on the delta table!
+
+    Get the highest previously read partition from the delta table.
+    Truncate all rows from the delta table that come from that partition.
+    Extract all rows from the EventHubCaptureExtractor from that partition on.
+    """
+
+    def __init__(
+        self, eh: EventHubCaptureExtractor, dh: DeltaHandle, dataset_key: str = None
+    ):
+        super().__init__(dataset_key=dataset_key)
+        self.eh = eh
+        self.dh = dh
+
+    def _read_pdate_partitioned(self) -> DataFrame:
+        """get the highest pdate partition,
+        truncate it,
+        read the event hub from that partition"""
+        max_pdate: Optional[dt] = (
+            self.dh.read().groupBy().agg(f.max("pdate")).collect()[0][0]
+        )
+        if max_pdate is None:
+            # if it is None, no previous data exists,
+            # so don't truncate and read everything
+            return self.eh.read()
+
+        max_pdate = max_pdate.astimezone(timezone.utc)
```

Comment:
make tests of this part! Right now the only tests is when max_pdate is None.

Reply:
tests added